fakestorage/upload_test.go

```diff
@@ -389,6 +389,50 @@ func TestServerClientObjectOperationsFailureToWriteExistingObject(t *testing.T)
 	})
 }
 
+func TestServerClientUploadRacesAreOnlyWonByOne(t *testing.T) {
+	const (
```

Comment:
Can you use `runServersTest` to make sure we're testing both backends?

Reply:
Sure! I worried a little bit about test-performance, given that this test is trying to provoke a race, therefor running multiple iterations. However, I think perhaps we could tune down `repetitions` slightly to compensate. Attaching test-durations from my machine below.

Also pushing a small improvement in test-diagnostics. On conflict, the test will now describe the conflict in more detail.

ok  	github.com/fsouza/fake-gcs-server	0.016s
ok  	github.com/fsouza/fake-gcs-server/fakestorage	1.176s
ok  	github.com/fsouza/fake-gcs-server/internal/backend	0.031s
ok  	github.com/fsouza/fake-gcs-server/internal/checksum	0.010s
ok  	github.com/fsouza/fake-gcs-server/internal/config	0.010s
ok  	github.com/fsouza/fake-gcs-server/internal/notification	0.026s